### PR TITLE
Log formats and general improvements

### PR DESCRIPTION
**Pull request by initstring**
This PR will try to help those working with tooling output programmatically.

It gets rid of the old `printc` function and replaces it with `fmt_output`. This consumes findings as python dictionaries, giving us flexibility in how we deal with the data.

It behaves the same in terms of printing to the screen. However, when a log file is specified it can write the individual items to it either in CSV or JSON.

Still to do:

- [x] Add a severity value (probably based on public exposure)
- [x] Update the deb package manpage stuff
- [ ] Tag a new version after merging

Ideas from: https://github.com/initstring/cloud_enum/issues/46

UPDATE:

I'm taking the time to address some other things in this PR, too:

- Lots of linting
- Some additional error checks
- Dealing with some edge-case DNS failures
- More accurate reporting of GCP AppEngine when behind auth

### CODE REVIEW

File: cloud_enum.py [hunk 1]
```diff
@@ -69,6 +67,8 @@ def parse_arguments():
 
     parser.add_argument('-l', '--logfile', type=str, action='store',
                         help='Will APPEND found items to specified file.')
+    parser.add_argument('-f', '--format', type=str, action='store', default='json', 
+                        help='Format for log file (json or csv, defaults to json)')
```
Comment (prryplatypus) [inline cloud_enum.py:71]: To avoid breaking changes (unless you don't mind), maybe it's worth making this default to `None` and if that's the case then save the output in the current, existing format?

Comment (initstring) [reply cloud_enum.py]: Hmmm, good point. I'm not so concerned about breaking what was already a hard to parse log file. But perhaps it's worth adding a "text" option as the default that logs it as plain text, without the ansi stuff. Some might be more familiar with that.

Comment (initstring) [reply cloud_enum.py]: Done!
